fluentd/lib/fluent-plugin-viaq_data_model/lib/fluent/plugin/viaq_data_model_log_level_normalizer.rb

```diff
@@ -82,10 +83,24 @@ def normalize_level!(record, priority=nil)
             record['level'] = retlevel || 'unknown'
         end
 
-        # extract_level_from_message evaluates a message against a regex expression
-        # that returns the first named group.  The named group is assumed to be
-        # prefixed with the pattern: 'l[0-9]_' or nil if unmatched
+        # extract_level_from_message evaluates:
+        # - if it 'Hash'and has 'level' field and return if existed
+        # - if it JSON valid string, try to parse it and return 'level' if exist
+        # - a message  against a regex expression
+        #   that returns the first named group.  The named group is assumed to be
+        #   prefixed with the pattern: 'l[0-9]_' or nil if unmatched
         def extract_level_from_message(message, check)
+            if message.is_a?(Hash)
+                if message.key?('level')
+                    return message['level']
+                end
+            end    
+            if valid_json?(message)
```

Comment:
this will force every message to be parsed as json, even if json parsing is not enabled via the CLO api. This will impact performance, please reconsider.